.dotnet/src/Custom/Assistants/FileSearchToolDefinition.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace OpenAI.Assistants;
+
+[CodeGenModel("AssistantToolsFileSearch")]
+[CodeGenSuppress(nameof(FileSearchToolDefinition))]
+public partial class FileSearchToolDefinition : ToolDefinition
+{
+    public int? MaxResults
+    {
+        get => _fileSearch.InternalMaxNumResults;
+        init => _fileSearch.InternalMaxNumResults = value;
+    }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="FileSearchToolDefinition"/>.
+    /// </summary>
+    public FileSearchToolDefinition()
+        : this("file_search", new ChangeTrackingDictionary<string, BinaryData>(), new InternalAssistantToolsFileSearchFileSearch())
```

Comment:
The only piece of code that should ever call the constructor that sets the serialized additional raw data is serialization code. If you feel like you have to call it because there is no other constructor that does what you want, that is typically a sign that you need to add a new constructor.